### PR TITLE
Fix: Add tags to SQS when value is an empty string

### LINKED ISSUE #4343: bug: Localstack drops empty string tag value and causes ElasticMQ exception
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If I run the following Terraform configuration against Localstack with ElasticMQ backend, then the resource creation hangs while ElasticMQ throws exceptions.

```terraform
terraform {
  required_providers {
    aws = "3.50.0"
  }
}

provider "aws" {
  region                      = "eu-west-1"

  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true

  endpoints {
    sqs            = "http://localhost:4566"
  }
}

resource "aws_sqs_queue" "queue" {
  name  = "foobar"

  tags = {
    NoFoo = ""
  }
}
```

If I run `ncat -l localhost 4566` instead of Locastack, I can see the `Tag.1.Value=` is there.

```
POST / HTTP/1.1
Host: localhost:4566
User-Agent: APN/1.0 HashiCorp/1.0 Terraform/0.12.31 (+https://www.terraform.io) terraform-provider-aws/3.50.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.39.5 (go1.16; linux; amd64)
Content-Length: 382
Authorization: AWS4-HMAC-SHA256 Credential=mock_access_key/20210722/eu-west-1/sqs/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=1a535809020db386c13b4b4169308bd79d81f47edb3922c9a59c4707d7b51061
Content-Type: application/x-www-form-urlencoded; charset=utf-8
X-Amz-Date: 20210722T133950Z
Accept-Encoding: gzip
Connection: close

Action=CreateQueue&Attribute.1.Name=DelaySeconds&Attribute.1.Value=0&Attribute.2.Name=MaximumMessageSize&Attribute.2.Value=262144&Attribute.3.Name=MessageRetentionPeriod&Attribute.3.Value=345600&Attribute.4.Name=ReceiveMessageWaitTimeSeconds&Attribute.4.Value=0&Attribute.5.Name=VisibilityTimeout&Attribute.5.Value=30&QueueName=foobar&Tag.1.Key=NoFoo&Tag.1.Value=&Version=2012-11-05
```

If I exec into the Localstack container and attach `tshark` on ElasticMQ's port then I can see the query param got dropped.
(`tshark -i lo -f "tcp port 52097" -e http.file_data -Tjson`)

```
"Action=CreateQueue&Attribute.1.Name=DelaySeconds&Attribute.1.Value=0&Attribute.2.Name=MaximumMessageSize&Attribute.2.Value=262144&Attribute.3.Name=MessageRetentionPeriod&Attribute.3.Value=345600&Attribute.4.Name=ReceiveMessageWaitTimeSeconds&Attribute.4.Value=0&Attribute.5.Name=VisibilityTimeout&Attribute.5.Value=30&QueueName=foobar&Tag.1.Key=NoFoo&Version=2012-11-05"
```

### Expected Behavior

The SQS queue should be created without error and the tag set with empty value (`""`). 

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run --rm -it -p "4566:4566" -e "SERVICES=sqs" -e "DEBUG=1" -e "SQS_PROVIDER=elasticmq" localstack/localstack:0.12.15

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    terraform init && terraform apply -auto-approve // on the above Terraform snippet put in main.tf

### Environment

```markdown
- OS: Ubuntu 20.04
- LocalStack: 0.12.15
- Terraform: 1.0.2
```

### Anything else?

As ElasticMQ could handle this edge case more gracefully (at least respond with HTTP 400 and not HTTP 500) I also opened in issue there:  https://github.com/softwaremill/elasticmq/issues/511

### PR DESCRIPTION
**Pull request by mgagliardo**
Fixes #4343. 

*Note:* __Tests will not pass until  https://github.com/spulec/moto/pull/4087 is merged__

When passing an empty string as tag to SQS this is the `req_data` passed to the `CreateQueue` function:

```
{'Action': 'CreateQueue', 'Attribute.1.Name': 'DelaySeconds', 'Attribute.1.Value': '0', 'Attribute.2.Name': 'MaximumMessageSize', 'Attribute.2.Value': '262144', 'Attribute.3.Name': 'MessageRetentionPeriod', 'Attribute.3.Value': '345600', 'Attribute.4.Name': 'ReceiveMessageWaitTimeSeconds', 'Attribute.4.Value': '0', 'Attribute.5.Name': 'VisibilityTimeout', 'Attribute.5.Value': '30', 'QueueName': 'foobar', 'Tag.1.Key': 'NoFoo', 'Version': '2012-11-05'}
```

The querystring actually has no value at all:
```
Action=CreateQueue&Attribute.1.Name=DelaySeconds&Attribute.1.Value=0&Attribute.2.Name=MaximumMessageSize&Attribute.2.Value=262144&Attribute.3.Name=MessageRetentionPeriod&Attribute.3.Value=345600&Attribute.4.Name=ReceiveMessageWaitTimeSeconds&Attribute.4.Value=0&Attribute.5.Name=VisibilityTimeout&Attribute.5.Value=30&QueueName=foobar&Tag.1.Key=NoFoo&Version=2012-11-05
```

So there's a missing `Tag.1.Value`, now if we loop through every `Tag.N` index and check if there's  a present k/v `Value` for that (which the only case for this is passing an empty string as value). Also extending integration tests for this.

Terraform output with this fix:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_sqs_queue.queue will be created
  + resource "aws_sqs_queue" "queue" {
      + arn                               = (known after apply)
      + content_based_deduplication       = false
      + deduplication_scope               = (known after apply)
      + delay_seconds                     = 0
      + fifo_queue                        = false
      + fifo_throughput_limit             = (known after apply)
      + id                                = (known after apply)
      + kms_data_key_reuse_period_seconds = (known after apply)
      + max_message_size                  = 262144
      + message_retention_seconds         = 345600
      + name                              = "foobar"
      + name_prefix                       = (known after apply)
      + policy                            = (known after apply)
      + receive_wait_time_seconds         = 0
      + tags                              = {
          + "NoFoo" = ""
        }
      + tags_all                          = {
          + "NoFoo" = (known after apply)
        }
      + url                               = (known after apply)
      + visibility_timeout_seconds        = 30
    }

Plan: 1 to add, 0 to change, 0 to destroy.
aws_sqs_queue.queue: Creating...
aws_sqs_queue.queue: Creation complete after 0s [id=http://localhost:4566/000000000000/foobar]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

And AWS CLI get tags:
```
$ awslocal sqs list-queue-tags --queue-url http://localhost:4566/000000000000/foobar
{
    "Tags": {
        "NoFoo": ""
    }
}
```

### CODE REVIEW

**Comment by mgagliardo** [comment]
Waiting on merge from https://github.com/spulec/moto/pull/4254 and then we can pull that same version here, should (finally, hopefully) fix this issue

**Comment by mgagliardo** [comment]
> Waiting on merge from [spulec/moto#4254](https://github.com/spulec/moto/pull/4254) and then we can pull that same version here, should (finally, hopefully) fix this issue

CC @whummer the PR got merged and once this pull the latest moto lib to our own this should pass without any issues. Let me know if there's anything else I need to do here. Thanks!

File: tests/integration/test_sqs.py [hunk 1]
```diff
@@ -58,6 +59,7 @@
 
 THIS_FOLDER = os.path.dirname(os.path.realpath(__file__))
 TEST_LAMBDA_ECHO_FILE = os.path.join(THIS_FOLDER, "lambdas", "lambda_echo.py")
+TEST_LAMBDA_TAGS = {"tag1": "value1", "tag2": "value2", "tag3": ""}
```
Comment (whummer) [inline tests/integration/test_sqs.py:62]: nitpick: Could probably be renamed to `TEST_TAGS`, as these tags do not seem to be related to Lambdas in this case. (not critical, though)..

File: localstack/services/sqs/sqs_listener.py [hunk 2]
```diff
@@ -457,6 +461,21 @@ def get_message_attributes_md5(cls, req_data):
...
+    # Fixes tags with empty strings as value
+    def fix_missing_tag_values(self, req_data):
+        keys_matched = []
+        for k, v in req_data.items():
+            match = re.match(r"^Tag\.(\d+)\.Key", k)
+            if match:
+                index = match.group(1)
+                tag_val = "Tag.{}.Value".format(index)
+                if tag_val not in req_data.keys():
+                    keys_matched.append(tag_val)
```
Comment (whummer) [inline localstack/services/sqs/sqs_listener.py:473]: nit: I guess we could remove `keys_matched` entirely, and simply assign the default value here?
```
                if tag_val not in req_data.keys():
                    req_data[tag_val] = ""
```

**Review by whummer — APPROVED** [review]
Looks great, thanks for extending the logic and tests for this @mgagliardo  👍 (nitpick comments are not critical - maybe we can pick them up in a future iteration..)
